http-server-netty/src/main/java/io/micronaut/http/server/netty/websocket/NettyServerWebSocketUpgradeHandler.java

```diff
@@ -275,4 +257,41 @@ public void channelInactive(@NonNull ChannelHandlerContext ctx) throws Exception
         super.channelInactive(ctx);
         cancelUpgrade = true;
     }
+
+    private static final class WebsocketRequestLifecycle extends RequestLifecycle {
+        @Nullable
+        final RouteMatch<?> route;
+
+        boolean shouldProceedNormally;
+
+        WebsocketRequestLifecycle(RouteExecutor routeExecutor, HttpRequest<?> request, @Nullable RouteMatch<?> route) {
+            super(routeExecutor, request);
+            this.route = route;
+        }
+
+        ExecutionFlow<MutableHttpResponse<?>> handle() {
+            MutableHttpResponse<?> proceed = HttpResponse.ok();
+
+            if (route != null) {
+                request().setAttribute(HttpAttributes.ROUTE_MATCH, route);
```

Comment:
We might want to have some `InternalHttpRequest` with `setRouteMatch` and `setRouteInfo` to treat these values as fields and avoid modifying the attributes map, with backward compatibility for `setAttribute/getAttribute`

Reply:
out of scope for this pr